src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java

```diff
@@ -100,6 +104,14 @@ public AuthorizationMatrixProperty(Map<Permission,? extends Set<String>> granted
             this.grantedPermissions.put(e.getKey(),new HashSet<>(e.getValue()));
     }
 
+    @DataBoundConstructor // JENKINS-49199: Used for job-dsl
+    @Restricted(NoExternalUse.class)
+    public AuthorizationMatrixProperty(List<String> permissions) {
```

Comment:
This could be a databound setter, not to tie you down to using a constructor and deprecating and all.

Reply:
I would not consider permissions optional (which is, at least in Pipeline, the defining difference between what goes into a constructor and what is just a setter), and since it's `@Restricted` I don't need to do the `@Deprecated` dance in the future either.